numpy/random/generator.pyx

```diff
@@ -3859,6 +3867,9 @@ cdef class Generator:
                     x[i] = buf
         else:
             # Untyped path.
+            if axis != 0:
+                raise NotImplementedError("Axis argument is only supported "
```

Comment:
Should this be documented here and/or for permutation?

Reply:
This doesn't apply to `permutation`, but would indeed be worth mentioning for `shuffle`

Reply:
@wkschwartz @eric-wieser Added one sentence in the doc of `axis`.